style_guide.rst

```diff
@@ -0,0 +1,393 @@
+     *  ERROR: A recoverable error
+     *  WARNING: A warning
+     *  INFO: High-level information about the progress of the application
+     *  DEBUG: Lower-level information
+     *  TRACE: Very low-level information.
+     *
+     */
+
+
+Function Comments
+^^^^^^^^^^^^^^^^^
+
+Function comments should be done in the form of a multiline comment
+above the function header.
+
+This comment must contain information specific to what a function does.
+It should also include a description of the purpose and expected
+input arguments, the expected output values, and how error conditions
+are handled.
+
+Example:
+
+::
+
+    /*
+     * chilog - Print a log message
+     *
+     * level: Logging level of the message
+     *
+     * fmt: printf-style formatting string
+     *
+     * ...: Extra parameters if needed by fmt
+     *
+     * Returns: nothing.
+     */
+    void chilog(loglevel_t level, char *fmt, ...);
+
+
+Block Comments
+^^^^^^^^^^^^^^
+
+Block comments (``/* ... */``) generally apply to some (or all) code that follows
+them, and are indented to the same level as that code.
+
+Inline Comments
+^^^^^^^^^^^^^^^
+
+Use inline comments sparingly.  An inline comment is a comment on the
+same line as a statement. Inline comments should be separated by at
+least two spaces from the statement.  They should start with a ``//`` and a
+single space.
+
+Inline comments are unnecessary and in fact distracting if they state
+the obvious. Don't do this:
+
+::
+
+   x = x + 1                 // Increment x
+
+But sometimes, this style of comment is useful:
+
+::
+
+   x = x + 1                 // Compensate for border
+
+
+
+Naming Conventions
+------------------
+
+Variable and function names should use the `snake_case <https://en.wikipedia.org/wiki/Snake_case>`_
+naming convention (i.e., ``lowercase_with_underscore``). For example:
+
+::
+
+          sum_of_squares
+          print_happy_birthday
+          total_apples
+
+
+Constants names should use snake_case with all caps:
+
+::
+
+    PI
+    MAX_CLIENTS
+    MAX_IRC_MSG_LEN
+
+
+Use descriptive names for parameter names, variables, and function
+names.  Use short names for local
+variables.  In general, the further away a variable will be used, the more
+descriptive the name needs to be.
+
+
+..
+    Printing logging / debug messages
+    ---------------------------------
+
+    All the projects in this class use a simple logging library called ``chilog`` that is documented in each of the project specifications. You must use the ``chilog`` functions *exclusively* for printing logging or debug messages. **Do not use printf() directly in your code**. Please note that the ``chilog`` functions provide essentially the same functionality as ``printf``, so there is no situation where ``printf`` would be necessary instead of ``chilog`` (using ``chilog`` consistently also means you will not have to scrub ``printf``'s from your code before submitting it).
```

Comment:
"that is documented in each of the project specifications" <- doesn't apply to cs220

Reply:
This section of the style guide is actually commented out for now. It's taken directly from Networks, and I need to come up with more general guidelines on proper logging / message printing.